File: API/TravelPics.Dashboard.API/TravelPics.Domains/Entities/Post.cs
namespace TravelPics.Domains.Entities
{
    public class Post
    {
        public int Id { get; set; }
        public virtual Location Location { get; set; }
        public string? Description { get; set; }
        public int CreatedById { get; set; }
        public virtual ICollection<Document> Photos { get; set; }
        public virtual User User { get; set; }
        public DateTimeOffset? PublishedOn { get; set; }
        public bool IsDeleted { get; set; }
        public Post()
        {
            Photos = new List<Document>();
        }
    }
}

File: TravelPics.UI/src/app/services/api/dtos/post.ts
import { Location } from "./location";
import { Image } from "./image";
import { User } from "./user";
import { Like } from "./like";

export interface Post{
    id: number;
    location: Location;
    description: string;
    photos: Image[];
    createdById: number;
    publishedOn: Date;
    user: User;
    likes: Like[];
}